docs/BackupAndRestore.md

```diff
@@ -0,0 +1,594 @@
+```
+
+If remote authentication is needed, `database.autoImport.credentials` should also be set in a form of `user:password`.
+
+The process of automatically downloading the remote source will require some time depending on the network speed and the size of the file.
+Import progress can be monitored by requesting the log of the desired SM pod.
+
+An example log indicating successful automatic archive import can be seen below.
+
+```
+2021-02-19T10:32:27.748+0000 ===========================================
+2021-02-19T10:32:27.754+0000 logsize=73; maxlog=5000000
+2021-02-19T10:32:27.765+0000 Created new dir /var/opt/nuodb/archive/nuodb/demo
+2021-02-19T10:32:33.185+0000 archiveId=-1; DB=demo; hostname=sm-database-nuodb-cluster0-demo-0
+2021-02-19T10:32:34.742+0000 path=/var/opt/nuodb/archive/nuodb/demo; atoms=0; catalogs=0
+2021-02-19T10:32:36.395+0000 Automatic archive import will be performed for archiveId=-1, source=http://nginx.web.svc.cluster.local/demo-20201223T132332-stream.tar.gz, type=stream, strip=1
+2021-02-19T10:32:36.441+0000 Restoring http://nginx.web.svc.cluster.local/demo-20201223T132332-stream.tar.gz; existing archive directores: total 4
+drwxr-xr-x 2 nuodb root 4096 Feb 19 10:32 demo
+2021-02-19T10:32:36.470+0000 (restore) recreated /var/opt/nuodb/archive/nuodb/demo; atoms=0
+2021-02-19T10:32:36.538+0000 curl -k  http://nginx.web.svc.cluster.local/demo-20201223T132332-stream.tar.gz | tar xzf - --strip-components 1 -C /var/opt/nuodb/archive/nuodb/demo
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+100 74801  100 74801    0     0  8116k      0 --:--:-- --:--:-- --:--:-- 8116k
+2021-02-19T10:32:36.650+0000 restoring archive and/or clearing restored archive physical metadata
+2021-02-19T10:32:38.387+0000 Finished restoring /var/opt/nuodb/archive/nuodb/demo to /var/opt/nuodb/archive/nuodb/demo. Created archive with archive ID 1
+...
+```
+
+### Automatic archive restore
+
+_Automatic archive restore_ is a special case of the seed restore operation.
+It allows SM to automatically restore its archive if it is corrupted or missing from a predefined restore source so that the accelerated sync use case is supported automatically.
+Although this mechanism can be used with a remote URL, it is typically used with `:latest` or `:group-latest` configured as a restore source.
+
+_Automatic archive restore_ is configured in the `database.autoRestore` section and will execute for every Storage Manager in the database deployment when all of the below conditions are met:
+
+- `database.autoRestore.source` and `database.autoRestore.type` are configured
+- archive metadata is found for the archive in the admin layer or on disk
+- configured restore source is available in the SM pod
+- the archive is detected as corrupted or missing - if the atom files count is less than 20 and catalog atom files count is less than 2
```

Comment:
Isn't it also a condition that the database has running SMs, otherwise this would be a full database restore? Also, how automated is this, i.e. does it require any user intervention whatsoever?

Reply:
There is no check on the database state. I assume that in the general case when no user storage groups are defined --- if we automatically restore the archive from an old backup set, then it won't be selected as leader candidate or the next database cold start.

The `autoRestore` is fully automated and is triggered by SM pod start/restart.